src/arithmetic/arithmetic_expression.c

```diff
@@ -298,8 +298,17 @@ static AR_EXP_Result _AR_EXP_EvaluateFunctionCall(AR_ExpNode *node, const Record
 	return res;
 }
 
-static inline void _AR_EXP_UpdateEntityIdx(AR_OperandNode *node, const Record r) {
-	node->variadic.entity_alias_idx = Record_GetEntryIdx(r, node->variadic.entity_alias);
+static inline AR_EXP_Result _AR_EXP_UpdateEntityIdx(AR_OperandNode *node, const Record r) {
```

Comment:
_AR_EXP_UpdateEntityIdx should return a different type, `AR_EXP_Result` doesn't make sense in this context.